src/pages/Animal/Feed/index.tsx

```diff
@@ -1,40 +1,128 @@
-import { useNavigation } from '@react-navigation/native';
+import React, { useLayoutEffect } from 'react';
 import { setStatusBarBackgroundColor } from 'expo-status-bar';
-import React, { useEffect, useLayoutEffect, useState } from 'react';
+
 import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
+
+import { useNavigation } from '@react-navigation/native';
+
 import { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
-import { v4 as uuidv4 } from 'uuid';
-import { AnimalCard } from '../../../components/AnimalCard';
-import { Theme } from '../../../constants';
+
 import HeaderLayout from '../../../layouts/HeaderLayout';
+
+import AnimalCard from '../../../components/AnimalCard';
+import InfiniteScroll from '../../../components/InfiniteScroll';
+
+import { Theme } from '../../../constants';
+
+import {
+  Age,
+  AgeNames,
+  Animal,
+  Sex,
+  SexNames,
+  Size,
+  SizeNames,
+} from '../../../types/animal';
+
+import { formatLocation } from '../../../utils/formatLocation';
+
 import {
   CardText, CardTextContainer, CardTextRow, Container,
 } from './styles';
 
 // Service imports.
 import animalAPI from '../../../services/animal/api';
+import userAPI from '../../../services/user/api';
 
 const FeedPets = (): JSX.Element => {
   const navigation = useNavigation();
 
-  const [fetchedPets, setFetchedPets] = useState<FirebaseFirestoreTypes.DocumentData[]>([]);
-
   useLayoutEffect(() => {
     setStatusBarBackgroundColor(Theme.elements.statusBarSecondaryDark, false);
   }, [navigation]);
 
-  const fetchPets = (): void => {
-    animalAPI.getAll()
-      .then((result) => {
-        const data = result.docs.map((doc) => ({ id: uuidv4(), ...(doc.data()) }));
-        setFetchedPets(data);
+  const animalKey = (animalItem: Animal): string => animalItem.id;
+
+  const formatAnimal = (pet: Animal): JSX.Element => (
+    <AnimalCard
+      key={pet?.id}
+      imageUrl={`${pet?.pictures.length > 0 ? `${pet?.pictures[0]}` : null}`}
```

Comment:
Question: Why use `${}`? Couldn't this code be written just as plain JS?

Reply:
Solved